File: Fintracker.Domain/Entities/Wallet.cs
using Fintracker.Domain.Common;

namespace Fintracker.Domain.Entities;

public class Wallet : FinancialEntity
{
    public Wallet()
    {
        Users = new HashSet<User>();
        Transactions = new HashSet<Transaction>();
        Budgets = new HashSet<Budget>();
    }

    public User Owner { get; set; } = default!;
    public Guid OwnerId { get; set; }

    public ICollection<User> Users { get; set; }

    public ICollection<Transaction> Transactions { get; set; }

    public ICollection<Budget> Budgets { get; set; }
    

    public bool IsBanking { get; set; }

    public string? BankAccountId { get; set; } = default!;
}

File: Fintracker.UI/src/entities/Wallet.ts
import {RegisterOptions} from "react-hook-form";
import {Budget} from "./Budget.ts";
import { Transaction } from "./Transaction.ts";
import { FinancialEntity } from "./FinancialEntity.ts";
import { User } from "./User.ts";

export interface Wallet extends FinancialEntity {
    ownerId: string,
    isBanking: boolean;
    bankAccountId: string;
    owner: User;
    budgets: Budget[];
    transactions: Transaction[];
    users: User[];
}

export const nameRegisterOptionsForWallet: RegisterOptions = {
    maxLength: {value: 50, message: "Maximum name for wallet is 50"},
    minLength: {value: 3, message: "Minimum length for wallet name is 3"},
    required: "Name for wallet is required",
}

export const balanceRegisterOptionsForWallet: RegisterOptions = {
    valueAsNumber: true,
    max: {value: 100_000_000_000, message: "Maximum start balance for wallet is 100 billions"},
    required: 'Balance is required'
}